src/core/config/ProviderSettingsManager.ts

```diff
@@ -80,9 +80,19 @@ export class ProviderSettingsManager {
 	constructor(context: ExtensionContext) {
 		this.context = context
 
-		// TODO: We really shouldn't have async methods in the constructor.
-		this.initialize().catch(console.error)
+		// kilocode_change start
+		// only initialize ONCE, and save the promise in case somebody needs to wait.
+		this.initialization = this.init_runMigrations()
 	}
+	private readonly initialization: Promise<void>
```

Comment:
```suggestion
		// kilocode_change start
		// only initialize ONCE, and save the promise in case somebody needs to wait.
		this.initialization = this.init_runMigrations()
		// kilocode_change end
	}

	// kilocode_change start
	private readonly initialization: Promise<void>
```

Reply:
I mean sure, but why is that better?

Reply:
if you put them visually farther apart, it is less likely the markers are even noticed in case of a merge conflict.